pom.xml

```diff
@@ -206,6 +207,26 @@
                 </configuration>
             </plugin>
 
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-antrun-plugin</artifactId>
+                <version>3.1.0</version>
+                <executions>
+                    <execution>
+                        <phase>generate-resources</phase>
+                        <configuration>
+                            <target>
+                                <echo message="compilerFactory=${shadeCalciteUnescaped}.org.codehaus.janino.CompilerFactory"
+                                      file="src/main/resources/${shadeCalciteUnescaped}.org.codehaus.commons.compiler.properties"/>
```

Comment:
Maven best practice is to generate files into the build directory (typically `target`), but then we would need yet another plugin to add that resource directory to the build.

Instead of generating this file, what if we simply add the file to `src/main/resources` and check it in?

Reply:
```diff
diff --git a/src/main/resources/com.linkedin.coral.calcite.$internal.org.codehaus.commons.compiler.properties b/src/main/resources/com.linkedin.coral.calcite.$internal.org.codehaus.commons.compiler.properties
new file mode 100644
index 0000000..9586dc0
--- /dev/null
+++ b/src/main/resources/com.linkedin.coral.calcite.$internal.org.codehaus.commons.compiler.properties
@@ -0,0 +1 @@
+compilerFactory=com.linkedin.coral.calcite.$internal.org.codehaus.janino.CompilerFactory

Reply:
That works too, just want to make sure we have everything in one place so we don't forget to update the property file in case the prefix is changed. Please let me know if the adding the file directly is preferred and I will make a new revision.